src/node/txdownloadman_impl.cpp

```diff
@@ -361,83 +366,47 @@ node::RejectedTxTodo TxDownloadManagerImpl::MempoolRejectedTx(const CTransaction
 
     if (state.GetResult() == TxValidationResult::TX_MISSING_INPUTS) {
         // Only process a new orphan if this is a first time failure, as otherwise it must be either
-        // already in orphanage or from 1p1c processing.
+        // already in orphanage or from 1p1c processing. Also, we are unable to process orphans that are too low feerate
+        // right now, as package validation only permits one generation of CPFP.
         if (first_time_failure && !RecentRejectsFilter().contains(ptx->GetWitnessHash().ToUint256())) {
-            bool fRejectedParents = false; // It may be the case that the orphans parents have all been rejected
-
             // Deduplicate parent txids, so that we don't have to loop over
             // the same parent txid more than once down below.
             unique_parents = GetUniqueParents(tx);
 
-            // Distinguish between parents in m_lazy_recent_rejects and m_lazy_recent_rejects_reconsiderable.
-            // We can tolerate having up to 1 parent in m_lazy_recent_rejects_reconsiderable since we
-            // submit 1p1c packages. However, fail immediately if any are in m_lazy_recent_rejects.
-            std::optional<Txid> rejected_parent_reconsiderable;
-            for (const Txid& parent_txid : unique_parents) {
-                if (RecentRejectsFilter().contains(parent_txid.ToUint256())) {
-                    fRejectedParents = true;
-                    break;
-                } else if (RecentRejectsReconsiderableFilter().contains(parent_txid.ToUint256()) &&
```

Comment:
I get that if this is an orphan, we only have the parent txids. In most cases, it seems like this reconsiderable filter will be storing wtxid? The exceptions being a non-segwit tx and if a parent is TX_RECONSIDERABLE but is actually witness-stripped (I'm not sure if there are others?). I may be missing something here, are there more cases where this reconsiderable filter check be used with txid?

Reply:
Yep, we don't ever put txids in the reconsiderable filter unless the txid is the same as the wtxid. We don't put witness-stripped in there either (even if we do #32379, as that will put it in the normal RecentRejects filter)

Reply:
I modified p2p_opportunistic_1p1c.py to remove the witness on a low-fee parent tx and it failed with TX_RECONSIDERABLE (as those checks are before the 3x CheckInputScripts that returns WITNESS_STRIPPED). I think my comment was a little confusing, also not sure this matters in practice?